summingbird-core/src/main/scala/com/twitter/summingbird/memory/Memory.scala

```diff
@@ -82,11 +82,11 @@ class Memory(implicit jobID: JobId = JobId("default.memory.jobId")) extends Plat
           case AlsoProducer(l, r) =>
             //Plan the first one, but ignore it
             val (left, leftM) = toStream(l, jamfs)
+            val (right, rightM) = toStream(r, leftM)
```

Comment:
here is the bug in Memory producer. You should make the `lforcedEmpty` lazy or be sure to plan the right before you evaluate it. Here I do both since it seemed more correct to plan as much as possible before causing side effects, but also that we don't want side-effects at all until the job is run.

The underlying issue is that `Stream` is not entirely lazy. The head is eager. So, if you do `.filter(_ => false)` there will be no head but it evaluates as much as possible until it finds a head, which means it evaluates the whole stream. By evaluating the steam it hits the side effects in the map functions on `sumByKey` and `write`.